enhancements/baremetal/continuous-hw-introspection.md

```diff
@@ -0,0 +1,114 @@
+the capability to perform hardware inspection on each of the servers periodically and update
+the "status" field of the BareMetalHost CRD. Please refer to [1] for the latest BaremetalHost
+CRD.
+
+## Motivation
+
+This enhancement will allow the user to get continuous updates on all the baremetal
+servers in their Metal3 deployment providing useful metrics about the deployment and
+monitor the health of the baremetal servers.
+
+### Goals
+
+This is enhancement will allow the baremetal host CR associated with each baremetal
+server in a metal3 deployment to contain up-to-date informtaion about the HW and
+operational status of a server.
+
+### Non-Goals
+
+This proposal does not include any changes to the BareMetalHost CRD.
+
+## Proposal
+
+This enhancement proposes to add a new Daemonset started by the machine-api-operator
+only for the BareMetal platform. This Daemonset would run a new container image
+(ironic-hardware-inventory-recorder) on all master and worker nodes. This new service
+would keep the Hardware status on each of these baremetal servers updated by continuously
+gathering HW parameters on each of these baremetal nodes.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The machine-api-operator would now be responsible for deploying the Daemonset that would
+run the service that performs continuous HW introspection on baremetal hosts.The container
+image for the Daemonset can be found at [2]. Also, if for some reason, a previously "ready"
+metal3 deployment becomes "unavailable", the daemonset can continue to keep running.
+This just will result in the lastest state of the baremetal servers in the baremetal CRs
+eventually become stale until metal3 transitions to "available"  operational state again.
+
+### Risks and Mitigations
+
+No technical risks forseen for this proposal.
+
+## Design Details
+
+### Test Plan
+
+Any e2e tests would make sure that the new Daemonset is running on all the master
+and worker nodes in a Baremetal IPI deployment. The tests would make sure that
+the HW status expressed in the BareMetalHost CR is updated periodically indicating
+that the different components of the underlying service are communicating with each other.
+Lastly, any e2e tests will make sure that the Daemonset is running even if "metal3"
+is not.
+
+### Graduation Criteria
+
+[TODO] This feature is intended to be in tech preview in 4.5.
+
+### Upgrade / Downgrade Strategy
+
+Upgrade to 4.5 and downgrade from 4.5 are both non-issues because there is no change to the
+BareMetalHost CRD to implement this proposal. After a downgrade, the HW introspection data
+will eventually become stale.
+
+### Version Skew Strategy
+
+## Implementation History
+
+[1] - https://github.com/openshift/machine-api-operator/blob/master/install/0000_30_machine-api-operator_08_baremetalhost.crd.yaml
+[2] - https://quay.io/repository/openshift/origin-ironic-hardware-inventory-recorder
```

Comment:
We could link to https://github.com/metal3-io/metal3-docs/blob/master/design/hardware-status.md for more detail.

Reply:
Done.